nibabel/openers.py

```diff
@@ -60,13 +67,31 @@ def readinto(self, buf):
             return n_read
 
 
-def _gzip_open(fileish, *args, **kwargs):
-    gzip_file = BufferedGzipFile(fileish, *args, **kwargs)
+def _gzip_open(fileish, mode='rb', compresslevel=9):
+
+    # is this a file? if not we assume it is a string
+    is_file = hasattr(fileish, 'read') and hasattr(fileish, 'write')
+
+    # If we've been given a file object, we can't change its mode.
```

Comment:
Does this mean we silently ignore the passed mode in this case?  Raise error instead maybe?  Or error if `mode is not None`?

Reply:
Yes - currently if `fileish` is a file-like object, the mode (if passed in) is ignored. 

We can't raise an error `if mode is not None` because its default value is `'rb'` - the `Opener.__init__` will set the `mode` argument if it is not provided by the caller.

How about I change the above behaviour, so that `Opener` passes `mode` through without changing/setting it, and then in `_gzip_open` do something like this:

```python
def _gzip_open(..., mode=None, ...):
    if hasattr(fileish, 'mode') and mode is None:
        mode = fileish.mode
    if hasattr(fileish, 'mode') and fileish.mode != mode:
        raise ValueError('Open file mode does not match requested mode')
```

Reply:
Actually, I think this is a problem which doesn't exist ... `_gzip_open` only ever gets called for file names - if `Opener.__init__` is passed an open file handle, it just stores it as-is, and returns.

So we don't need to worry about this at all :)